subsys/bluetooth/audio/mcs.c

```diff
@@ -808,6 +809,49 @@ static void notify(const struct bt_uuid *uuid, const void *data, uint16_t len)
 	}
 }
 
+/* Helper function to notify UTF8 string values
+ * Will truncate string to fit within notification if required.
+ * The string must be null-terminated.
+ */
+static void notify_string(const struct bt_uuid *uuid, const char *str)
+{
+	/* TODO:
+	 * This function will need to get the ATT_MTU to know what length to
+	 * truncate the string to.  But the ATT_MTU is per connection, and MCS
+	 * is not connection-aware yet.
+	 * For now: Truncate according to the default ATT_MTU, so that
+	 * notifications will go through
```

Comment:
I have a feeling this should be handled by the stack to send as much as it can depending on MTU.
So once `bt_gatt_notify_uuid` is called it should feed the buffer up to MTU and leave the client with the decision whether it wants to read the rest. Returning error is not sufficient since, the value should be notified anyway, and the Server has no power to exchange the MTU by itself

Reply:
I disagree. I think it should be up to the individual service determine if a notification should be truncated or not. 

Furthermore, as this PR also demonstrates, we may not always want to naively truncate e.g. UTF8 encoded strings either, as that would essentially send improperly encoded strings.

Reply:
I think Emil is correct.  What to do with notifications may be service-specific, and we do not want to put that kind of information into the stack.
- service-specific information does not belong in the general stack
- it is complicated to get it there - the notify would also have to include some kind of type information

In this case, what is truncated is track titles (later also player names), and those can well serve their purpose even if truncated - they will typically just be shown to the user.

Reply:
OK, you're correct